new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/AppBuilder.java

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.actions.impl;
+
+import java.io.IOException;
+
+import oracle.weblogic.kubernetes.actions.impl.primitive.Command;
+
+import static oracle.weblogic.kubernetes.actions.ActionConstants.APP_DIR;
+import static oracle.weblogic.kubernetes.actions.ActionConstants.ARCHIVE_DIR;
+import static oracle.weblogic.kubernetes.actions.impl.primitive.Command.defaultCommandParams;
+import static oracle.weblogic.kubernetes.utils.FileUtils.checkDirectory;
+import static oracle.weblogic.kubernetes.utils.FileUtils.cleanupDirectory;
+import static oracle.weblogic.kubernetes.utils.FileUtils.copyFolder;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.assertj.core.api.Assertions.fail;
+
+/**
+ *  Implementation of actions that build an application archive file.
+ */
+
+public class AppBuilder {
+  private static final String ARCHIVE_SRC_DIR = ARCHIVE_DIR + "/wlsdeploy/applications";
+  
+  private AppParams params;
+
+  /**
+   * Create an AppParams instance with the default values.
+   * @return an AppParams instance 
+   */
+  public static AppParams defaultAppParams() {
+    return new AppParams().defaults();
+  }
+
+  /**
+   * Set up the AppBuilder with given parameters.
+   * @return the AppBuilder instance 
+   */
+  public static AppBuilder withParams(AppParams params) {
+    return new AppBuilder().params(params);
+  }
+
+  private AppBuilder params(AppParams params) {
+    this.params = params;
+    return this;
+  }
+
+  /**
+   * Build an application archive using a pre-populated AppParams instance.
+   * @return true if the command succeeds 
+   */
+  public boolean build() {
+    // prepare the archive directory and copy over the app src
+    Throwable throwable = catchThrowable(
+        () -> { 
+        cleanupDirectory(ARCHIVE_DIR);
+        checkDirectory(ARCHIVE_SRC_DIR);
+        copyFolder(
+            APP_DIR + "/" + params.srcDir(), 
+            ARCHIVE_SRC_DIR);
+        });
+        
+    assertThat(throwable)
```

Comment:
assertions are for tests, when implementing an action, i think we can just use the normal try/catch 
@rjeberhard thoughts?

Reply:
i don't like tests relying on assertions hidden away in actions/helper methods

Reply:
Agreed.  The test should deal with the exception.  Let "build()" throw it.

Reply:
Removed all assertion in the actions.